File: java-packages/src/main/java/com/rmaciak/order/domain/OrderItem.java
package com.rmaciak.order.domain;

import java.math.BigDecimal;
import java.util.UUID;

final class OrderItem {

    OrderItem(UUID id, BigDecimal quota) {
        this.id = id;
        this.quota = quota;
    }

    private final UUID id;
    private final BigDecimal quota;

    BigDecimal itemQuota() {
        return quota;
    }

    UUID itemId() {
        return id;
    }
}

File: kotlin-with-submodules/order-domain/src/main/kotlin/com/rmaciak/order/OrderItem.kt
package com.rmaciak.order

import java.math.BigDecimal
import java.util.UUID

internal data class OrderItem(
        private val id: UUID,
        private val quota: BigDecimal
) {

    internal fun itemQuota(): BigDecimal = quota

    internal fun toEntity() =
            OrderItemEntity(
                    id = this.id,
                    quota = this.quota
            )

    internal fun toDto() =
            OrderItemDto(
                    id = this.id,
                    quota = this.quota
            )
}